compiler+runtime/src/cpp/jank/runtime/obj/persistent_string_sequence.cpp

```diff
@@ -0,0 +1,93 @@
+#include <jank/runtime/obj/persistent_string_sequence.hpp>
+
+namespace jank::runtime
+{
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr const s)
+    : str{ s }
+  {
+    assert(!s->data.empty());
+  }
+
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr const s, size_t const i)
+    : str{ s }
+    , index{ i }
+  {
+    assert(!s->data.empty() && i < s->data.size());
+  }
+
+  /* behavior::objectable */
+  native_bool obj::persistent_string_sequence::equal(object const &o) const
+  {
+    return detail::equal(o, str->data.begin(), str->data.end());
+  }
+
+  void obj::persistent_string_sequence::to_string(fmt::memory_buffer &buff) const
+  {
+    return behavior::detail::to_string(str->data.begin() + index, str->data.end(), "(", ')', buff);
+  }
+
+  native_persistent_string obj::persistent_string_sequence::to_string() const
+  {
+    fmt::memory_buffer buff;
+    behavior::detail::to_string(str->data.begin() + index, str->data.end(), "(", ')', buff);
+    return { buff.data(), buff.size() };
+  }
+
+  native_hash obj::persistent_string_sequence::to_hash() const
+  {
+    return hash::ordered(str->data.begin(), str->data.end());
```

Comment:
Bug here: missing `+ index`. This will mean `(= (hash (next (seq "foo"))) (hash (seq "foo")))`, which is incorrect.